File: codility/task-1/Driver.java

class Solution {

  public int solution (String S) {
    int length = S.length();
    // Such index only exists if the length of the string is odd
    if (length % 2 == 0) {
      return -1;
    }

    int middle = length / 2;
    for (int i = middle - 1, j = middle + 1; i >= 0 && j < length; i--, j++) {
      if (S.charAt(i) != S.charAt(j)) {
        return -1;
      }
    }

    return middle;
  }
}

/**
 * @author shengmin
 */
public class Driver {
  public static void main(String[] args) {
    Solution sol = new Solution();
    System.out.println(sol.solution(""));
    System.out.println(sol.solution("a"));
    System.out.println(sol.solution("aa"));
    System.out.println(sol.solution("aba"));
    System.out.println(sol.solution("abc"));
    System.out.println(sol.solution("racecar"));

  }
}

File: leetcode/candy/Driver.scala
object Driver {
  def main(a: Array[String]) {
    val s = new Solution()
    println(s.candy(Array(1, 0, 2)))
    println(s.candy(Array(1, 2, 3, 4, 5, 6, 5, 4, 3, 2, 1)))
    println(s.candy(Array(5, 5, 4, 3, 3, 5, 8, 9, 7)))
  }
}
